flink-connectors/flink-connector-hive/src/test/java/org/apache/flink/connectors/hive/HiveDialectITCase.java

```diff
@@ -807,6 +807,29 @@ public void testShowPartitions() throws Exception {
         assertTrue(partitions.toString().contains("dt=2020-04-30 01:02:03/country=china"));
     }
 
+    @Test
+    public void testStatementVariableSubstitution() {
+        // test system variable for substitution
+        System.setProperty("k1", "v1");
+        List<Row> result =
+                CollectionUtil.iteratorToList(
+                        tableEnv.executeSql("select '${system:k1}'").collect());
+        assertEquals("[+I[v1]]", result.toString());
+
+        // test env variable for substitution
+        String classPath = System.getenv("CLASSPATH");
```

Comment:
env `CLASSPATH ` maybe not exist.
We should test for an environment variable that exists and an environment variable that does not exist